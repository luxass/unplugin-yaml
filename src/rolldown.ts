/**
 * This entry file is for Rolldown plugin.
 *
 * @module
 */

import type unplugin from "./";
import { createRolldownPlugin } from "unplugin";
import { unpluginFactory } from "./";

/**
 * Rolldown plugin
 *
 * @example
 * ```ts
 * // rolldown.config.js
 * import yaml from "unplugin-yaml/rolldown"
 *
 * export default {
 *   plugins: [yaml()],
 * }
 * ```
 */
export default createRolldownPlugin(unpluginFactory) as typeof unplugin.rolldown;
