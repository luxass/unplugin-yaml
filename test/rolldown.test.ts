import { join } from "node:path";
import { rolldown } from "rolldown";
import { describe, expect, it } from "vitest";
import { testdir } from "vitest-testdirs";
import YAMLPlugin from "../src/rolldown";
import { removeComments } from "./utils";

describe("handles yaml", () => {
  it("expect yaml import to be a json object", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/basic/yaml"));

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "basic.js"),
      plugins: [
        YAMLPlugin(),
      ],
    });
    const { output } = await bundle.generate({
      format: "esm",
      sourcemap: false,
    });

    expect(removeComments(output[0].code)).toMatchSnapshot();
  });

  it("expect yaml import to be a string", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/basic/yaml"));

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "basic-raw.js"),
      plugins: [
        YAMLPlugin(),
      ],
    });
    const { output } = await bundle.generate({
      format: "esm",
      sourcemap: false,
    });

    expect(removeComments(output[0].code)).toMatchSnapshot();
  });
});

describe("handle yml", () => {
  it("expect yml import to be a json object", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/basic/yml"));

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "basic.js"),
      plugins: [
        YAMLPlugin(),
      ],
    });
    const { output } = await bundle.generate({
      format: "esm",
      sourcemap: false,
    });

    expect(removeComments(output[0].code)).toMatchSnapshot();
  });

  it("expect yml import to be a string", async () => {
    const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/basic/yml"));

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "basic-raw.js"),
      plugins: [
        YAMLPlugin(),
      ],
    });
    const { output } = await bundle.generate({
      format: "esm",
      sourcemap: false,
    });

    expect(removeComments(output[0].code)).toMatchSnapshot();
  });
});

it("handle multi document", async () => {
  const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/multi"));

  expect(testdirPath).toBeDefined();

  const bundle = await rolldown({
    input: join(testdirPath, "multi.js"),
    plugins: [
      YAMLPlugin({
        type: "multi",
      }),
    ],
  });
  const { output } = await bundle.generate({
    format: "esm",
    sourcemap: false,
  });

  expect(removeComments(output[0].code)).toMatchSnapshot();
});

it("handle transforms", async () => {
  const testdirPath = await testdir.from(join(import.meta.dirname, "fixtures/transform"));

  expect(testdirPath).toBeDefined();

  const bundle = await rolldown({
    input: join(testdirPath, "transform.js"),
    plugins: [
      YAMLPlugin({
        transform(data) {
          if (data != null && typeof data === "object" && "this" in data) {
            return {
              this: "transformed",
            };
          }
        },
      }),
    ],
  });
  const { output } = await bundle.generate({
    format: "esm",
    sourcemap: false,
  });

  expect(removeComments(output[0].code)).toMatchSnapshot();
});
