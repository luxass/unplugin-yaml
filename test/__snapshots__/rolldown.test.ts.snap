// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`expect yaml import to be a string 1`] = `
"
var yaml_config_default = "pluginDir: ./plugins\\n\\nweb:\\n  enabled: true\\n\\nlogging:\\n  type: stdout\\n  level: info\\n";

console.log(yaml_config_default);

"
`;

exports[`expect yml import to be a string 1`] = `
"
var yml_config_default = "pluginDir: ./plugins\\n\\nweb:\\n  enabled: true\\n\\nlogging:\\n  type: stdout\\n  level: info\\n";

console.log(yml_config_default);

"
`;
